### DISCUSSION TITLE
Query.prototype.update with option `overwrite: true` does not work with arrays

### DISCUSSION THREAD

**Discussion by eliw00d** [Automattic/mongoose#10274] [Q&A]
I have the following pre findOneAndUpdate hook:

```javascript
schema.pre('findOneAndUpdate', function() {
  const filter = this.getFilter()
  const update = this.getUpdate()
  const options = this.getOptions()

  const document = await this.findOne(filter).exec()

  // I use set to update to the document and then use toObject to transform according to some schema versioning switch cases
  const object = document?.set(update).toObject()

  // I want to be able to overwrite if it's an update, otherwise it is an upsert
  await this.update(filter, object ?? update, { ...options, overwrite: !!object })
})
```

And the schema has a toObject.transform function that handles schema versioning. 

What I am trying to do is overwrite the document to get rid of some fields when old fields are no longer required. I have an array of objects type and when that gets set, the result of this update is an empty array, even though a new array was provided. 

Is there a way to get this to work with overwrite?

**Comment by vkarpov15** [comment]
@eliw00d without more information on what "I have an array of objects type and when that gets set, the result of this update is an empty array, even though a new array was provided." means in terms of code and observed+expected behavior, it would be hard to provide any meaningful advice.
